### PR TITLE
3257 cleanup old auth part 4: Remove old controllers and tests from subsystems

### PR DESCRIPTION
**Pull request by sahma19**
Part 4 of: [https://app.zenhub.com/workspaces/team-atlas-633199659e255a37cd1d144f/issues/gh/energinet-datahub/energy-origin-issues/3257](https://app.zenhub.com/workspaces/team-atlas-633199659e255a37cd1d144f/issues/gh/energinet-datahub/energy-origin-issues/3257)

### CODE REVIEW

File: domains/authorization/Proxy.IntegrationTests/Setup/ProxyTestAuthHandler.cs [hunk 1]
```diff
@@ -21,25 +19,38 @@ protected override Task<AuthenticateResult> HandleAuthenticateAsync()
     {
         var jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
 
-        if (!Context.Request.Headers.ContainsKey(HeaderNames.Authorization))
+        if (!Context.Request.Headers.ContainsKey(HeaderNames.Authorization) ||
+            string.IsNullOrWhiteSpace(Context.Request.Headers[HeaderNames.Authorization]))
```
Comment (TopSwagCode) [inline domains/authorization/Proxy.IntegrationTests/Setup/ProxyTestAuthHandler.cs:23]: string.IsNullOrWhiteSpace(Context.Request.Headers[HeaderNames.Authorization]

if (!authorizationHeader.StartsWith("Bearer "))

 are both checked by the framework / oidc package. Don't really need to check it twice.

File: domains/certificates/Query.API/API.IntegrationTests/Factories/IntegrationTestB2CAuthHandler.cs [hunk 2]
```diff
@@ -21,20 +22,39 @@ public IntegrationTestB2CAuthHandler(IOptionsMonitor<AuthenticationSchemeOptions
     protected override Task<AuthenticateResult> HandleAuthenticateAsync()
     {
         var jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
-        var authorizationHeader = Context.Request.Headers[HeaderNames.Authorization].ToString().Substring(7);
-        var securityToken = jwtSecurityTokenHandler.ReadToken(authorizationHeader) as JwtSecurityToken;
 
-        // Check for the existence of a claim that only B2C should provide
-        if (!securityToken!.Claims.Any(c => c.Type == ClaimType.OrgIds))
```
Comment (TopSwagCode) [inline domains/certificates/Query.API/API.IntegrationTests/Factories/IntegrationTestB2CAuthHandler.cs:28]: Why is this removed? We require that claim orgids to be there.

File: domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs [hunk 3]
```diff
@@ -127,7 +114,15 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
                 services.Remove(services.First(s => s.ServiceType == typeof(Meteringpoint.V1.Meteringpoint.MeteringpointClient)));
                 services.AddSingleton(MeteringpointClient);
             }
+
+            services.AddAuthentication(options =>
```
Comment (TopSwagCode) [inline domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs:118]: Did our old test not test the new auth?

File: domains/certificates/Query.API/API/ContractService/Clients/WalletClient.cs [hunk 4]
```diff
@@ -133,15 +133,6 @@ private void ValidateOwnerAndSubjectMatch(string owner)
                 throw new HttpRequestException("Owner must match subject");
             }
         }
-        else
```
Comment (TopSwagCode) [inline domains/certificates/Query.API/API/ContractService/Clients/WalletClient.cs:136]: Maybe remove if "IsBearerTokenIssuedByB2C()". Because now all tokens should be issued by it ?

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs [hunk 5]
```diff
@@ -14,21 +15,23 @@
...
 {
-    [Authorize(Policy = PolicyName.RequiresCompany)]
     [HttpPost("start")]
     [ProducesResponseType(typeof(ClaimAutomationArgumentDto), 201)]
     [ProducesResponseType(typeof(ClaimAutomationArgumentDto), 200)]
-    public async Task<ActionResult> StartClaimAutomation()
+    public async Task<ActionResult> StartClaimAutomation([FromQuery] Guid organizationId)
     {
-        var user = new UserDescriptor(User);
+        if (!accessDescriptor.IsAuthorizedToOrganization(organizationId))
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs:29]: Just general question. Was if (!accessDescriptor.IsAuthorizedToOrganization(organizationId)) the thing we wanted all places that uses organization id? Can't really remember where we ended on this. @martinhenningjensen @sahma19 .

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs [hunk 6]
```diff
@@ -14,21 +15,23 @@
...
 namespace API.ClaimAutomation.Api.Controllers;
 
-[Authorize]
 [ApiController]
-[ApiVersion(ApiVersions.Version20240103)]
+[Authorize(Policy.Frontend)]
+[ApiVersion(ApiVersions.Version20240515)]
 [Route("api/claim-automation")]
-public class ClaimAutomationController(IUnitOfWork UnitOfWork) : ControllerBase
+public class ClaimAutomationController(IUnitOfWork unitOfWork, AccessDescriptor accessDescriptor) : ControllerBase
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs:22]: Did we forget to migrate this ?

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs [hunk 7]
```diff
@@ -6,39 +6,40 @@
...
 using FluentValidation.AspNetCore;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace API.Transfer.Api.Controllers;
 
-[Authorize]
 [ApiController]
-[ApiVersion(ApiVersions.Version20240103)]
+[Authorize(Policy.Frontend)]
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:18]: Did we forget to migrate this aswell ? :S

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs [hunk 8]
```diff
@@ -6,39 +6,40 @@
...
+    [HttpPost]
     [ProducesResponseType(typeof(TransferAgreementProposalResponse), 201)]
     [ProducesResponseType(typeof(void), 400)]
     [ProducesResponseType(typeof(void), 409)]
-    [HttpPost]
-    public async Task<ActionResult> CreateTransferAgreementProposal(CreateTransferAgreementProposal request)
+    public async Task<ActionResult> CreateTransferAgreementProposal([FromQuery] Guid organizationId, CreateTransferAgreementProposal request)
     {
-        var user = new UserDescriptor(User);
+        accessDescriptor.AssertAuthorizedToAccessOrganization(organizationId);
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:42]: Is this the way we should check if user has right and not if statement?

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs [hunk 9]
```diff
@@ -106,14 +115,14 @@ public async Task<ActionResult<TransferAgreementProposalResponse>> GetTransferAg
             return NotFound();
         }
 
-        var user = new UserDescriptor(User);
+        accessDescriptor.AssertAuthorizedToAccessOrganization(organizationId);
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:118]: I would put this as first line in api endpoint. No need to ask database if you don't have access.

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs [hunk 10]
```diff
@@ -150,29 +160,29 @@ public async Task<ActionResult> DeleteTransferAgreementProposal(Guid id)
...
```
Comment (TopSwagCode) [inline domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:163]: Same as before. I would put this first. SECURITY FIRST?!?!?!?! MAKE SECURITAAAA GRAET AGAIN! :P

**Comment by TopSwagCode** [comment]
Looks great. Awesome work. Most comments are more general questions.

File: domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs (hunk 3 above)
Comment (sahma19) [reply domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs:118]: apparantly not no.

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs (hunk 6 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs:22]: No. This is just how git shows diffs when you:

1. have two files named A and B.
2. Delete A
3. Rename B to A

Then it looks like what you see now.

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs (hunk 7 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:18]: No. This is just how git shows diffs when you:

have two files named A and B.
Delete A
Rename B to A
Then it looks like what you see now.

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs (hunk 8 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:42]: I do not know. I did not change anything

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs (hunk 9 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:118]: I just deleted old code, and renamed the controller. that is all

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs (hunk 10 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:163]: It was there before. did not change anything

File: domains/certificates/Query.API/API/ContractService/Clients/WalletClient.cs (hunk 4 above)
Comment (sahma19) [reply domains/certificates/Query.API/API/ContractService/Clients/WalletClient.cs:136]: Fixed: https://github.com/Energinet-DataHub/energy-origin/pull/1484/commits/202a1d87d4955a337d121327a5bcf101ad8dfd30

File: domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs (hunk 5 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API/ClaimAutomation/Api/Controllers/ClaimAutomationController.cs:26]: I can't remember.

File: domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs (hunk 3 above)
Comment (TopSwagCode) [reply domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs:120]: :(

File: domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs (hunk 8 above)
Comment (TopSwagCode) [reply domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs:43]: It's fine :D Just saw it's transfer that is special.

**Review by TopSwagCode — APPROVED** [review]
